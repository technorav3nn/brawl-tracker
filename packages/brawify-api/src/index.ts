export * from "./lib/handlers";
export * from "./lib/types";
