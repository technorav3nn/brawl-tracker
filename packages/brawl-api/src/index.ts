export { FetchError as BrawlApiFetchError } from "ofetch";

export * as brawlApi from "./lib/handlers";
export * from "./lib/types";

const a = {
	b: 3,
};
