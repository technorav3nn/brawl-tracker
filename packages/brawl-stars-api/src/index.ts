export * from "./lib/client";
export * from "./lib/errors";

export * from "./lib/types/api";
export * from "./lib/types/brawlers";
export * from "./lib/types/powers";
export * from "./lib/types/events";
export * from "./lib/types/battlelogs";
export * from "./lib/types/players";
export * from "./lib/types/clubs";
export * from "./lib/types/rankings";
