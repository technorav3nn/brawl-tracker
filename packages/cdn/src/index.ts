import { $fetch } from "ofetch";
import type { BrawlerData } from "./lib/types/brawler";

const CDN_URL = "https://cdn.deathblows.tech/file/bs-api/assets";

export function normalizeNameToCdnName(name: string) {
	// "El Primo" -> "el_primo" for example
	return name.toLowerCase().replaceAll(" ", "_");
}

/**
 * Get brawler data from CDN
 *
 * @param name - Normalized brawler name
 */
export async function getBrawlerData(name: string) {
	return $fetch<BrawlerData>(`${CDN_URL}/brawlers/${name}/data.json`);
}

/**
 * Get brawler avatar URL from CDN
 *
 * @param name - Normalized brawler name
 */
export function getBrawlerAvatarUrl(name: string) {
	return `${CDN_URL}/brawlers/${name}/avatar.png`;
}

/**
 * Get brawler model URL from CDN
 *
 * @param name - Normalized brawler name
 */
export function getBrawlerModelUrl(name: string) {
	return `${CDN_URL}/brawlers/${name}/model.png`;
}
