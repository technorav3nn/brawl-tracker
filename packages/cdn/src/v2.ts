import { $fetch } from "ofetch";
import { type CdnBrawler } from "./lib/types/brawler-v2";
import { type GearData } from "./lib/types/gears";

export const CDN_URL_V2 = "https://cdn.deathblows.dev";

export * from "./lib/types/brawler-v2";
export * from "./lib/types/gears";

/**
 * Get brawler data from CDN
 *
 * @param id - Normalized brawler name
 */
export async function getBrawlerData(id: string) {
	return $fetch<CdnBrawler>(`${CDN_URL_V2}/brawlers/${id}/data.json`);
}

export async function getAllGears() {
	return $fetch<GearData>(`${CDN_URL_V2}/gears/all-info.json`);
}

/**
 * Get brawler avatar URL from CDN
 *
 * @param id - Brawler ID
 */
export function getBrawlerPortraitUrl(id: string) {
	return `${CDN_URL_V2}/brawlers/${id}/portrait.webp`;
}

/**
 * Get brawler model URL from CDN
 *
 * @param id - Brawler ID
 */
export function getBrawlerModelUrl(id: string) {
	return `${CDN_URL_V2}/brawlers/${id}/model.webp`;
}

export * from "./lib/types/brawler";
