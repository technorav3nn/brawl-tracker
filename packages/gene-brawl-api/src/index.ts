import { $fetch, FetchError } from "ofetch";
import type { ApiProfile, ApiProfileResponse, ApiUnresolvedProfileStats, ResolvedProfileStats } from "./lib/types/profile";
import type { ShopResponse } from "./lib/types/shop";

export * from "./lib/types/profile";
export * from "./lib/types/shop";

const BASE_URL = "https://api.hpdevfox.ru";

export async function getShop() {
	return $fetch<ShopResponse>(`${BASE_URL}/shop`);
}

export async function getPlayerProfile(tag: string) {
	const data = await $fetch<ApiProfileResponse<ApiUnresolvedProfileStats>>(`${BASE_URL}/profile/${tag}`);
	if (data?.detail === 0) {
		throw new FetchError("Invalid player tag");
	}

	(data.response.Stats as unknown as ApiProfile<ResolvedProfileStats>["Stats"]) = resolveStats(data.response.Stats);
	return data as unknown as ApiProfileResponse<ResolvedProfileStats>;
}

function resolveStats(stats: ApiUnresolvedProfileStats): ResolvedProfileStats {
	return {
		"3v3Wins": stats[1],
		duoWins: stats[11],
		soloWins: stats[8],
		trophies: stats[3],
		bossFightMaxDifficulty: stats[12],
		roboRumble: stats[9],
		powerPlayHighestRank: stats[14],
		cityRampageMaxDifficulty: stats[16],
		challengeWins: stats[15],
		brawlersOwned: stats[5],
		fame: stats[20],
		clubLeague: stats[19],
		currentRankedRank: stats[23],
		highestRankedRank: stats[22],
		experiencePoints: stats[2],
		highestTrophies: stats[4],
		legacyExperiencePoints: stats[21],
		powerPlayHighest: stats[13],
		powerLeagueSolo: stats[18],
		powerLeagueTeam: stats[17],
	};
}
