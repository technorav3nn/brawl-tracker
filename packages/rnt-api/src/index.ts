import { $fetch, type $Fetch, type FetchOptions } from "ofetch";
import { RNT_API_BASE_URL } from "./lib/constants";
import { AlliancesRoute } from "./lib/routes/alliances";
import { ProfileRoute } from "./lib/routes/profile";

export class Client {
	private readonly $fetch: $Fetch;

	public alliances: AlliancesRoute;
	public profiles: ProfileRoute;

	public constructor(fetchOptions: FetchOptions = {}) {
		this.$fetch = $fetch.create({
			baseURL: RNT_API_BASE_URL,
			...fetchOptions,
		});
		this.alliances = new AlliancesRoute(this.$fetch);
		this.profiles = new ProfileRoute(this.$fetch);
	}
}

export * from "./lib/types/alliance-search";
export * from "./lib/structures/alliance-search/alliance-search-aliance";
export * from "./lib/structures/alliance-search/alliance-search-result";

export * from "./lib/types/profile";
export * from "./lib/structures/profile/profile-result";
