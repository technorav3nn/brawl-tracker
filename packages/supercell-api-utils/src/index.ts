/**
 * Copyright 2023 Clashperk and its contributors
 *
 * Use of this source code is governed by an MIT-style
 * license that can be found in the LICENSE file or at
 * https://github.com/clashperk/clashofclans.js/blob/main/LICENSE.
 *
 * Source: {@link https://github.com/clashperk/clashofclans.js/blob/main/src/util/Util.ts}
 */

/**
 * Convert a date string from API to a Date object
 *
 * @see {@link https://github.com/clashperk/clashofclans.js/blob/main/src/util/Util.ts#L84}
 * @param date - date string from the API
 * @returns The date as a Date object
 */
export function convertApiDate(date: string) {
	const yearMonthDay = `${date.slice(0, 4)}-${date.slice(4, 6)}-${date.slice(6, 8)}`;
	const hoursMinutesSeconds = `${date.slice(9, 11)}:${date.slice(11, 13)}:${date.slice(13)}`;
	return new Date(`${yearMonthDay}T${hoursMinutesSeconds}`);
}

/**
 * Fixes tags to be in the correct format
 *
 * @see {@link https://github.com/clashperk/clashofclans.js/blob/main/src/util/Util.ts#L33}
 * @param tag - The tag to format
 * @returns string
 */
export function formatTag(tag: string) {
	// eslint-disable-next-line unicorn/prefer-string-replace-all
	return `#${tag.toUpperCase().replace(/O/g, "0").replace(/^#/g, "").replace(/\s/g, "")}`;
}

/**
 * Verifies a tag is valid
 *
 * @see {@link https://github.com/clashperk/clashofclans.js/blob/main/src/util/Util.ts#L46}
 * @param tag - The tag to verify
 * @returns boolean
 */
export function verifyTag(tag: string) {
	// eslint-disable-next-line unicorn/better-regex
	return /^#?[0289PYLQGRJCUV]{3,}$/.test(tag);
}
