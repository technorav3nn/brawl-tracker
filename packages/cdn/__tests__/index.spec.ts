import { describe, expect, test } from "vitest";
import { getBrawlerData, normalizeNameToCdnName } from "../src";

describe("@brawltracker/cdn", () => {
	describe("getBrawlerData", () => {
		test("GIVEN valid brawler name THEN returns brawler data", async () => {
			const promise = getBrawlerData(normalizeNameToCdnName("El Primo"));
			await expect(promise).resolves.not.toThrow();

			const data = await promise;
			expect(data).toMatchSnapshot();
			expect(data.id).toBe("el_primo");
			expect(data.name).toBe("El Primo");
		});
	});
});
