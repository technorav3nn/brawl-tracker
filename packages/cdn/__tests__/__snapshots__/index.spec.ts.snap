// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@brawltracker/cdn > getBrawlerData > GIVEN valid brawler name THEN returns brawler data 1`] = `
{
  "attack": {
    "description": "El Primo fires off a furious flurry of four fiery fists. That's a spicy jalapeno knuckle sandwich!",
    "name": "Fists of Fury",
    "stats": {
      "attackbullets": "4",
      "attackrange": "3 (Short)",
      "attackspeed": "3261",
      "attacksupercharge": "8.46%, 9.306% (with Super Charge Gear)",
      "attackwidth": "1",
      "reload": "0.8 seconds (Very Fast)",
    },
    "statsByLevel": [
      {
        "list": [
          360,
          396,
          432,
          468,
          504,
          540,
          576,
          612,
          648,
          684,
          720,
        ],
        "name": "Damage per punch",
      },
    ],
  },
  "avatar": {
    "path": "/brawlers/el_primo/avatar.png",
  },
  "description": "El Primo is always in the spotlight and that's right where he wants to be! There's no audience he can't win over, except maybe his fellow performers...",
  "directory": "/brawlers/el_primo/",
  "gadgets": [
    {
      "description": "El Primo grabs the closest enemy within his reach and flips them like a pancake over his broad shoulders.",
      "id": "23000264",
      "name": "Suplex Supplement",
      "path": "/gadgets/23000264.png",
    },
    {
      "description": "El Primo summons a small meteor to strike the nearest enemy. It deals 2880 damage and destroys walls.",
      "id": "23000292",
      "name": "Asteroid Belt",
      "path": "/gadgets/23000292.png",
    },
  ],
  "healthByLevel": [
    6000,
    6600,
    7200,
    7800,
    8400,
    9000,
    9600,
    10200,
    10800,
    11400,
    12000,
  ],
  "history": [
    {
      "date": "03/07/17:",
      "description": "El Primo's attack shape was changed from a cone to a rectangle.",
      "type": "Neutral",
    },
    {
      "date": "03/07/17:",
      "description": "The El Rey Primo skin was added.",
      "type": "Neutral",
    },
    {
      "date": "11/08/17:",
      "description": "El Primo's health was increased to 1400 (from 1300).",
      "type": "Buff",
    },
    {
      "date": "04/09/17:",
      "description": "El Primo's rarity was changed to Common (from Rare).",
      "type": "Neutral",
    },
    {
      "date": "07/12/17:",
      "description": "El Primo's El Fuego Star Power was added.",
      "type": "Neutral",
    },
    {
      "date": "07/12/17:",
      "description": "The health and damage statistics of all Brawlers were multiplied by 4.",
      "type": "Neutral",
    },
    {
      "date": "18/12/17:",
      "description": "El Primo's health was increased to 5800 (from 5600).",
      "type": "Buff",
    },
    {
      "date": "18/12/17:",
      "description": "El Primo's El Fuego Star Power burn damage was increased to 600 (from 400).",
      "type": "Buff",
    },
    {
      "date": "18/12/17:",
      "description": "El Primo's El Fuego Star Power burn duration was decreased to 4 seconds (from 5).",
      "type": "Nerf",
    },
    {
      "date": "22/12/17:",
      "description": "El Primo's movement speed was increased to 700 (from 650).",
      "type": "Buff",
    },
    {
      "date": "16/01/18:",
      "description": "El Primo's health was increased to 6000 (from 5800).",
      "type": "Buff",
    },
    {
      "date": "27/01/18:",
      "description": "El Primo's health was decreased to 5800 (from 6000).",
      "type": "Nerf",
    },
    {
      "date": "09/03/18:",
      "description": "El Primo's health was decreased to 5600 (from 5800).",
      "type": "Nerf",
    },
    {
      "date": "21/03/18:",
      "description": "Every Brawler's movement speed was increased by 70 points, increasing El Primo's base movement speed to 770 (from 700).",
      "type": "Neutral",
    },
    {
      "date": "21/03/18:",
      "description": "All Brawlers' projectile speeds are increased by 9%.",
      "type": "Neutral",
    },
    {
      "date": "21/05/18:",
      "description": "El Primo's rarity was changed to Rare (from Common).",
      "type": "Neutral",
    },
    {
      "date": "29/05/18:",
      "description": "El Primo's main attack damage was increased to 340 (from 320) per punch.",
      "type": "Buff",
    },
    {
      "date": "19/06/18:",
      "description": "El Primo's main attack damage was increased to 360 (from 340) per punch.",
      "type": "Buff",
    },
    {
      "date": "19/06/18:",
      "description": "His health was increased to 5800 (from 5600).",
      "type": "Buff",
    },
    {
      "date": "29/01/19:",
      "description": "El Primo's El Fuego Star Power damage was increased to 800 (from 600).",
      "type": "Buff",
    },
    {
      "date": "29/01/19:",
      "description": "His flavor text was changed from "El Primo was hit by a meteor from outer space, giving him superhuman strength and unrivaled charisma." to "El Primo throws a flurry of punches at his enemies. His Super is a leaping elbow drop that deals damage to all caught underneath!",
      "type": "Neutral",
    },
    {
      "date": "15/04/19:",
      "description": "El Primo's main attack effect was retextured.",
      "type": "Neutral",
    },
    {
      "date": "14/08/19:",
      "description": "El Primo's Meteor Rush Star Power was added.",
      "type": "Neutral",
    },
    {
      "date": "29/08/19:",
      "description": "El Primo's El Fuego Star Power damage was increased to 1000 (from 800).",
      "type": "Buff",
    },
    {
      "date": "18/09/19:",
      "description": "El Primo's Meteor Rush Star Power speed boost was decreased to 28% (from 32%).",
      "type": "Nerf",
    },
    {
      "date": "18/09/19:",
      "description": "El Primo and his skins were remodeled. El Rudo Primo now costs 80 Gems (from 30) and El Rey Primo now costs 150 Gems (from 80).",
      "type": "Neutral",
    },
    {
      "date": "09/10/19:",
      "description": "El Primo's Meteor Rush Star Power duration was decreased to 4 seconds (from 5).",
      "type": "Nerf",
    },
    {
      "date": "23/10/19:",
      "description": "El Primo's El Fuego Star Power damage was increased to 1200 (from 1000).",
      "type": "Buff",
    },
    {
      "date": "23/10/19:",
      "description": "His Meteor Rush Star Power speed boost was decreased to 25% (from 28%).",
      "type": "Nerf",
    },
    {
      "date": "03/12/19:",
      "description": "The El Brown skin was added. It was featured as a Line Friends skin.",
      "type": "Neutral",
    },
    {
      "date": "19/02/20:",
      "description": "El Primo's health was increased to 6000 (from 5800).",
      "type": "Buff",
    },
    {
      "date": "17/03/20:",
      "description": "El Primo's Suplex Supplement Gadget was added.",
      "type": "Neutral",
    },
    {
      "date": "29/04/20:",
      "description": "El Primo’s Suplex Supplement Gadget can no longer throw Brawlers with invulnerability shields.",
      "type": "Nerf",
    },
    {
      "date": "02/07/20:",
      "description": "El Primo's True Silver and True Gold skins were added.",
      "type": "Neutral",
    },
    {
      "date": "19/08/20:",
      "description": "El Primo's Asteroid Belt Gadget was added.",
      "type": "Neutral",
    },
    {
      "date": "10/09/20:",
      "description": "The number of hits necessary to charge El Primo's Super was decreased to 9 hits (from 10).",
      "type": "Buff",
    },
    {
      "date": "10/09/20:",
      "description": "The number of Super hits to recharge his Super was decreased to 4 hits (from 5).",
      "type": "Buff",
    },
    {
      "date": "10/09/20:",
      "description": "His Asteroid Belt Gadget explosion radius was increased by 11%.",
      "type": "Buff",
    },
    {
      "date": "10/09/20:",
      "description": "Fixed an issue where El Primo's Asteroid Belt explosion effect was not matching the effective area of the explosion, as a result the visual explosion area has been increased by 26%.",
      "type": "Neutral",
    },
    {
      "date": "10/09/20:",
      "description": "The El Atómico Primo skin was added.",
      "type": "Neutral",
    },
    {
      "date": "17/09/20:",
      "description": "Fixed an issue which prevented the El Atómico skin from appearing in the Star Shop.",
      "type": "Neutral",
    },
    {
      "date": "15/03/21:",
      "description": "El Primo, El Rudo Primo, El Rey Primo, and El Atómico received facial animations.",
      "type": "Neutral",
    },
    {
      "date": "16/06/21:",
      "description": "El Primo gained a Trait where he can now charge his Super when receiving damage.",
      "type": "Buff",
    },
    {
      "date": "17/06/21:",
      "description": "The Super charge rate from his Trait was decreased to 20% (from 30%).",
      "type": "Nerf",
    },
    {
      "date": "09/07/21:",
      "description": "El Primo's Latin America Master League skins were added.",
      "type": "Neutral",
    },
    {
      "date": "25/08/21:",
      "description": "The number of hits necessary to charge El Primo's Super was increased to 16 hits (from 9).",
      "type": "Nerf",
    },
    {
      "date": "25/08/21:",
      "description": "The Super charge rate from his Trait is now based on El Primo's max health.",
      "type": "Neutral",
    },
    {
      "date": "23/09/21:",
      "description": "The El Dragón Verdoso and El Dragón Rosado skins were added.",
      "type": "Neutral",
    },
    {
      "date": "27/10/21:",
      "description": "El Primo's Suplex Supplement Gadget no longer bypasses Frank's Active Noise Cancelling Gadget.",
      "type": "Nerf",
    },
    {
      "date": "27/10/21:",
      "description": "El Primo's class was changed to Tank (from Heavyweight).",
      "type": "Neutral",
    },
    {
      "date": "16/12/21:",
      "description": "The number of hits necessary to charge El Primo's Super was decreased to 12 hits (from 16).",
      "type": "Buff",
    },
    {
      "date": "02/02/22:",
      "description": "The El Tigro skin was added. It was featured as a Year of the Tiger skin.",
      "type": "Neutral",
    },
    {
      "date": "13/02/22:",
      "description": "The El Corazón skin was added. It was featured as a Brawlentine's skin.",
      "type": "Neutral",
    },
    {
      "date": "01/03/22:",
      "description": "El Rudo Primo's and El Rey Primo's main attack and Super visual effects were reworked.",
      "type": "Neutral",
    },
    {
      "date": "01/03/22:",
      "description": "Fixed an issue with El Tigro's Super visual effects.",
      "type": "Neutral",
    },
    {
      "date": "12/12/22:",
      "description": "El Primo's Super damage was increased to 960 (from 800).",
      "type": "Buff",
    },
    {
      "date": "12/12/22:",
      "description": "El Corazón, El Dragón Verdoso, El Dragón Rosado, and El Tigro's main attack and Super visual effects were reworked.",
      "type": "Neutral",
    },
    {
      "date": "28/02/23:",
      "description": "El Primo, El Rudo Primo, El Rey Primo, and El Atómico's main attack visual effects were reworked.",
      "type": "Neutral",
    },
    {
      "date": "28/02/23:",
      "description": "El Primo's Super visual effects were reworked.",
      "type": "Neutral",
    },
    {
      "date": "06/03/23:",
      "description": "The El Mayordomo skin was added. It was featured as a Brawl Pass Season 17 exclusive skin.",
      "type": "Neutral",
    },
    {
      "date": "25/04/23:",
      "description": "El Corazón now costs 79 Gems (from 49).",
      "type": "Neutral",
    },
    {
      "date": "25/04/23:",
      "description": "El Atómico now costs 79 Gems (from 10000 Star Points).",
      "type": "Neutral",
    },
    {
      "date": "05/09/23:",
      "description": "All Brawlers' health and damage were increased from 5% to 10% with each Power Level.",
      "type": "Neutral",
    },
    {
      "date": "24/10/23:",
      "description": "El Primo's Asteroid Belt Gadget damage was increased to 2304 (from 2000) and now scales with Power Levels.",
      "type": "Neutral",
    },
    {
      "date": "24/10/23:",
      "description": "El Primo's El Fuego Star Power damage was increased to 1522 (from 1200) and now scales with Power Levels.",
      "type": "Neutral",
    },
    {
      "date": "24/10/23:",
      "description": "El Primo's flavor text was changed from "El Primo throws a flurry of punches at his enemies. His Super is a leaping elbow drop that deals damage to all caught underneath!" to "El Primo is always in the spotlight and that's right where he wants to be! There's no audience he can't win over, except maybe his fellow performers..."",
      "type": "Neutral",
    },
    {
      "date": "12/12/23:",
      "description": "Fixed an issue with El Atómico's Super visual effects.",
      "type": "",
    },
  ],
  "hypercharge": [],
  "id": "el_primo",
  "model": {
    "path": "/brawlers/el_primo/model.png",
  },
  "name": "El Primo",
  "skins": [
    {
      "directory": "/brawlers/el_primo/skins/El Primo/",
      "name": "El Primo",
      "skins": [
        {
          "name": "Default",
          "path": "/brawlers/el_primo/skins/El%20Primo/Default.png",
        },
        {
          "campaign": "Latin America Master League",
          "cost": "49 Gems",
          "name": "El Primo Universitario",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Primo%20Universitario.png",
        },
        {
          "campaign": "Latin America Master League",
          "cost": "49 Gems",
          "name": "El Primo Club Nacional",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Primo%20Club%20Nacional.png",
        },
        {
          "campaign": "Latin America Master League",
          "cost": "49 Gems",
          "name": "El Primo Corinthians",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Primo%20Corinthians.png",
        },
        {
          "cost": "79 Gems or 2750 Bling",
          "name": "El Rudo Primo",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Rudo%20Primo.png",
        },
        {
          "campaign": "Wasteland",
          "cost": "79 Gems or 2750 Bling",
          "name": "El Atómico",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20At%C3%B3mico.png",
        },
        {
          "campaign": "My Brawlentine",
          "cost": "79 Gems or 2750 Bling",
          "name": "El Corazón",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Coraz%C3%B3n.png",
        },
        {
          "cost": "149 Gems or 5000 Bling",
          "name": "El Rey Primo",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Rey%20Primo.png",
        },
        {
          "campaign": "Brown & Friends",
          "cost": "149 Gems",
          "name": "El Brown",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Brown.png",
        },
        {
          "campaign": "Once Upon",
          "cost": "149 Gems or 5000 Bling",
          "name": "El Dragón Verdoso",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Drag%C3%B3n%20Verdoso.png",
        },
        {
          "campaign": "Once Upon",
          "cost": "149 Gems or 5000 Bling",
          "name": "El Dragón Rosado",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Drag%C3%B3n%20Rosado.png",
        },
        {
          "campaign": "Lunar Brawl",
          "cost": "149 Gems",
          "name": "El Tigro",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Tigro.png",
        },
        {
          "campaign": "True Silver",
          "cost": "10000 Coins",
          "name": "True Silver",
          "path": "/brawlers/el_primo/skins/El%20Primo/True%20Silver.png",
        },
        {
          "campaign": "True Gold",
          "cost": "25000 Coins",
          "name": "True Gold",
          "path": "/brawlers/el_primo/skins/El%20Primo/True%20Gold.png",
        },
        {
          "campaign": "Brawl Pass",
          "cost": "Brawl Pass Exclusive",
          "name": "El Mayordomo",
          "path": "/brawlers/el_primo/skins/El%20Primo/El%20Mayordomo.png",
        },
      ],
    },
  ],
  "starpowers": [
    {
      "description": "Enemies caught in El Primo's Super will burn for 1692 damage over 4 seconds.",
      "id": "23000086",
      "name": "El Fuego",
      "path": "/starpowers/23000086.png",
    },
    {
      "description": "El Primo gains a 25% speed boost for 4.0 seconds after using his Super.",
      "id": "23000140",
      "name": "Meteor Rush",
      "path": "/starpowers/23000140.png",
    },
  ],
  "stats": {
    "class": "Tank",
    "movementspeed": "770 (Fast), 970 (with Meteor Rush), 1600 (with Super)",
    "rarity": "Rare",
    "voiceactor": "Alejandro "Alex" Sanchez",
  },
  "super": {
    "description": "Leaping high, El Primo drops an Intergalactic Elbow that knocks around enemies and destroys cover!",
    "name": "Flying Elbow Drop",
    "stats": {
      "superrange": "9 (Long)",
      "supersupercharge": "30%, 33% (with Super Charge Gear)",
    },
    "statsByLevel": [
      {
        "list": [
          960,
          1056,
          1152,
          1248,
          1344,
          1440,
          1536,
          1632,
          1728,
          1824,
          1920,
        ],
        "name": "Damage",
      },
    ],
  },
  "tips": [
    "El Primo can deal high damage but his main attack has one of the shortest ranges in the game. To attack long-ranged Brawlers, players must attempt to maneuver El Primo into range and dodge enemy fire. However, El Primo's Super allows him to travel a long distance, making it effective to get closer to enemies and attack them. Because of this, you can justify this strategy with your Trait. First, charge your Super with El Primo’s Trait by allowing the enemy to attack you, then once your Super is charged up, you can jump onto them and defeat them. Regardless of these tactics, El Primo almost always struggles without the presence or strategic use of walls.",
    "Take caution when facing Bull, as with a normal interaction, Bull can defeat El Primo due to his superior unload speed and range. You should try to attack Bull with the tip of El Primo's range; avoid getting too close or staying too far. Bull, when all his shells hit, can exactly three-shot an El Primo of the same Power Level, so avoiding some of his shells should secure a victory against him. El Primo can also use his Super to knock Bull back and maintain some distance, which allows El Primo to better secure a victory against Bull.",
    "El Primo is niche in Brawl Ball because of his high health and great mobility with his Super and fast movement speed. He also has the ability to self-pass if you have his Super charged up. Use your high health to get within Super range of the goal, kick the ball towards the goal, use your Super to jump to the ball, which will knock enemy defenders back and clear the goalpost, and then retrieve the ball and attempt to score. With his Meteor Rush Star Power, he can travel to the goal much faster after self-passing, which amplifies this strategy’s effectiveness. El Primo's Super is also effective in knocking the ball out of an enemy's grasp and dealing moderate damage. Both of El Primo's Gadgets can be put to use in this game mode. Suplex Supplement can be used to knock the ball out of an enemy's grasp, or to throw an enemy out of the way to score. Asteroid Belt is best used when trying to score, to break walls in front of the goal, weaken or defeat enemy defenders, and knock surviving defenders out of the way to score.",
    "In Brawl Ball, you must choose whether to use your Suplex Supplement Gadget or your Super wisely. To knock the ball away from an enemy regularly, use his Suplex Supplement Gadget. This is because it’s more reliable than his Super and will not potentially destroy your own obstacles. However, in overtime, his Super is more reliable as enemies are more likely to stay out of your Gadget’s range, and there aren’t any obstacles you’ll risk destroying if you are using your Super.",
    "El Primo is versatile in Heist because with his Super, he can jump onto the safe and start rapidly punching it without needing to travel around the safe's obstacles. His jump can also eliminate enemies or knock them back and destroy the obstacles around the safe, allowing other teammates to reach the safe with ease. He can deal higher damage with his El Fuego Star Power, dealing 1200 extra damage and his Asteroid Belt Gadget, dealing 2000 damage while also destroying more obstacles to assist allies.",
    "El Primo is useful in Gem Grab. He can knock or throw enemies back with his Super or Suplex Supplement Gadget, allowing for his team to control the Gem mine and collect Gems quickly. El Primo’s high health makes him also decent as the Gem Carrier, because he won’t be defeated easily while holding Gems, and his Super helps him escape from any approaching enemies.",
    "When his Super is used at close range, El Primo immediately hits the ground. This can be used to quickly cancel Frank or Carl's Supers, effectively incapacitating the opponent. While his Suplex Supplement Gadget is probably more useful in such situations, his Super has a variable range and his Gadget uses are limited, allowing his Super to be a good alternative when the moment is right.",
    "El Primo’s El Fuego Star Power has the ability to prevent enemies in his landing radius from healing for an additional 4 seconds, raising the required time for a Brawler to start healing from 3 seconds to 7 seconds. In addition, it effectively increases his Super damage from 1440 to 2640, guaranteeing enemies hit by his Super, with sufficiently low remaining health, to be defeated in the absence of external healing abilities.",
    "When playing Duo Showdown with a non-random teammate, it’s a good strategy to pair El Primo with long-ranged Brawlers like Colt or Brock to increase the chances of victory. El Primo can protect these Brawlers by tanking damage directed at them with his high health while they support him by attacking the enemy from a range.",
    "His Suplex Supplement can displace enemy Brawlers, stunning them and sending them back a fair distance. This is especially useful in Brawl Ball to remove enemy defenders from their goal, to halt approaching enemies from walking the ball to your own goal and scoring, or to quickly escape from enemies when you don’t have your Super. He can also use the Gadget to collect Siege Bolts by throwing them a fair distance away before they can collect the Bolt. If the conditions are right, El Primo can also use his Suplex Supplement Gadget to eliminate enemies with the poison gas in Showdown by standing near it so that he can then throw any nearby enemies into the gas.",
    "The asteroid from El Primo's Asteroid Belt Gadget attacks the Brawler closest to him. While enemies try to dodge the asteroid, their movement can be predictable, making them easy targets for him or his allies. It can also be used to hit an enemy taking cover behind obstacles, which opens up enemy cover. Keep in mind that an teammate of yours or El Primo himself might need specific obstacles to properly function regularly or on a certain map.",
    "El Primo does decent in Robo Rumble with his abilities. His piercing, high-damaging, and fast-reloading attack allow El Primo to consistently and effectively defend the safe. His fast movement speed, Meteor Rush Star Power, and Super also useful to catch up to robots and eliminate them quickly. His Suplex Supplement Gadget is very useful to throw enemies away from the safe, giving ample time to eliminate the robots.",
    "El Primo is outstanding in Hot Zone, as goes for all tanks. He can deal high damage to enemies inside the zones, but his Super is his most useful asset. You can use his Super to travel to zones faster, and his Super can deal high damage to enemies and knock them out of the zone. In addition, his Suplex Supplement Gadget can also throw enemies out of a zone, while Asteroid Belt can potentially force an enemy to leave the zone for fear of being damaged by the meteor, or by being knocked back by the landing of the meteor.",
    "El Primo is niche in Hold the Trophy. His high health and high damage output can be used to withstand a lot of damage while holding the trophy or to consistently defend the trophy holder. His Super and Suplex Supplement Gadget can be used knock or throw an enemy away from the trophy or steal it from the enemy trophy holder.",
    "You can try a strategy in bush maps in which you keep spamming your Suplex Supplement Gadget near bushes, and once an enemy comes near you, your Suplex Supplement Gadget will activate and throw the enemy back, surprising them in the process and giving you ample time to take advantage of this. Using this to determine their position and the Brawler being used, you can then plan your next move (attacking them or retreating). This especially works in Showdown (in maps like Dark Passage, NruhC nrevaC, and Cavern Churn).",
    "El Primo has a great synergy with healers like Poco or Byron, so while he’s hit by an enemy and receiving Super charge, his teammate can heal him. Once he charges his Super, he can use it on an enemy, significantly damage or eliminate the enemy, and repeat the process. A similar strategy is to use Ruffs' Field Promotion Star Power to increase El Primo’s maximum health and allow El Primo to withstand more damage while charging his Super, but this strategy is time-consuming and if El Primo dies, his health resets.",
    "El Primo's Suplex Supplement Gadget can cancel the attacks and Supers of Brawlers that have very slow unload speeds such as Frank. This prevents you from being stunned by Frank's Super and gives you a higher chance of surviving.",
  ],
  "url": "https://brawlstars.fandom.com/wiki/El_Primo",
  "voicelines": [],
}
`;
